### PR TITLE
[IFT] track design space intersection size in IntersectionInfo.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: incremental-font-transfer/src/patchmap.rs [hunk 1]
```diff
@@ -11,11 +11,13 @@ use std::io::Cursor;
...
 
+use font_types::Fixed;
 use font_types::Tag;
 
 use data_encoding::BASE64URL;
 use data_encoding_macro::new_encoding;
 
+use read_fonts::collections::RangeSet;
 use read_fonts::FontRead;
 use read_fonts::{
```
Comment (cmyr) [inline incremental-font-transfer/src/patchmap.rs:22]: once again the auto importing hurts me

**Review by cmyr — APPROVED** [review]
looks good.

I'm curious if you're enforcing the invariant that start <= end, for your ranges? And/or have some test coverage for this case?

**Comment by garretrieger** [comment]
> looks good.
> 
> I'm curious if you're enforcing the invariant that start <= end, for your ranges? And/or have some test coverage for this case?

Yeah, here: https://github.com/googlefonts/fontations/blob/main/incremental-font-transfer/src/patchmap.rs#L437

Also internally RangeSet ignores attempts to add ranges with start > end so it's not possible to construct one that has bad ranges.

**Comment by garretrieger** [comment]
> > looks good.
> > I'm curious if you're enforcing the invariant that start <= end, for your ranges? And/or have some test coverage for this case?
> 
> Yeah, here: https://github.com/googlefonts/fontations/blob/main/incremental-font-transfer/src/patchmap.rs#L437
> 
> Also internally RangeSet ignores attempts to add ranges with start > end so it's not possible to construct one that has bad ranges.

and tests here: https://github.com/googlefonts/fontations/blob/main/incremental-font-transfer/src/patchmap.rs#L1998

**Pull request merged**